### PR TITLE
Allow users to upload video via mux

### PR DESCRIPTION
**Pull request by mattcasey**
It's a lot of code, but most of it was heavily influenced by this example Next.js app: https://github.com/vercel/next.js/tree/canary/examples/with-mux-video

See a diagram of the flow from mux (titled "Uploading through a CMS UI"): https://www.mux.com/blog/video-uploads-with-mux#uploading-through-a-cms-ui

The flow:
1. The client calls: `POST /api/upload` to get an authenticated url for uploading the video
2. The client uploads large video files in chunks using the @mux/upchunk library. One it's done, we receive an Asset id and a playback id. Both are saved in the page's JSON content.
3. When viewing the video in prosemirror, the asset id is used to retrieve the upload status and also a signed token from the backend
4. Finally, pass the playback id to mux player, once video is ready

### CODE REVIEW

File: components/common/CharmEditor/components/ResizableImage.tsx [hunk 1]
```diff
@@ -1,13 +1,12 @@
-import type { NodeViewProps, RawSpecs } from '@bangle.dev/core';
-import { Plugin } from '@bangle.dev/core';
-import type { EditorState, EditorView, Node, Slice, Transaction } from '@bangle.dev/pm';
+import type { RawSpecs } from '@bangle.dev/core';
```
Comment (mattcasey) [inline components/common/CharmEditor/components/ResizableImage.tsx:1]: this was some cleanup to standardize image embeds, adding a 'delete' icon to make it easy to remove them

File: components/common/CharmEditor/components/ResizableImage.tsx [hunk 2]
```diff
@@ -27,29 +28,30 @@ const StyledEmptyImageContainer = styled(Box)`
...
-        <ImageIcon fontSize='small' />
-        <Typography>Add an image</Typography>
-      </StyledEmptyImageContainer>
-    </ListItem>
+    <BlockAligner onDelete={onDelete}>
+      <ListItem
+        button
+        disableTouchRipple
+        sx={{
+          backgroundColor: isSelected ? 'var(--charmeditor-active)' : theme.palette.background.light,
```
Comment (msadura) [inline components/common/CharmEditor/components/ResizableImage.tsx:43]: maybe we should have this color in our mui theme? `var(--charmeditor-active)` ?

File: pages/api/mux/upload.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+export type CreateUploadResponse = {
+  id: string;
+  url: string;
+};
+
+const handler = nc<NextApiRequest, NextApiResponse>({ onError, onNoMatch });
+
+handler.use(requireUser).post(createUploadEndpoint);
+
+async function createUploadEndpoint(req: NextApiRequest, res: NextApiResponse<CreateUploadResponse>) {
```
Comment (msadura) [inline pages/api/mux/upload.ts:22]: I often end up having a trouble of naming handlers for endpoints...  I really like "namespacing" function name with `Endpoint` suffix in  `createUploadEndpoint`  :D

File: pages/api/mux/asset/[id].ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+export type AssetRequest = {
+  pageId: string;
+  id: string;
+};
+
+export type AssetResponse = Asset;
+
+const handler = nc<NextApiRequest, NextApiResponse>({ onError, onNoMatch });
+
+handler.use(requireUser).get(getUploadEndpoint);
```
Comment (msadura) [inline pages/api/mux/asset/[id].ts:19]: `getUploadEndpoint` -> `getAssetEndpoint`

**Review by msadura — APPROVED** [review]
Awesome work on this!
TBH I had to read through carefully because the flow (or maybe function names?) is a bit tricky at first - "createUpload" that does not really upload, `getUpload` etc. :D but once you get mux flow it is all nice and clear :)

File: components/common/CharmEditor/components/ResizableImage.tsx (hunk 2 above)
Comment (mattcasey) [reply components/common/CharmEditor/components/ResizableImage.tsx:43]: this wasnt a change i added... but yea it probably needs some cleanup

File: pages/api/mux/asset/[id].ts (hunk 4 above)
Comment (mattcasey) [reply pages/api/mux/asset/[id].ts:19]: thanks

**Pull request merged**